google/cloud/aiplatform/tensorboard/uploader_main.py

```diff
@@ -0,0 +1,195 @@
+from tensorboard.plugins.hparams import metadata as hparams_metadata
+from tensorboard.plugins.image import metadata as images_metadata
+from tensorboard.plugins.graph import metadata as graphs_metadata
+
+from google.cloud import storage
+from google import auth as google_auth
+from google.auth.transport import grpc as google_auth_transport_grpc
+from google.auth.transport import requests as google_auth_transport_requests
+from google.cloud.aiplatform.tensorboard import uploader
+from google.cloud.aiplatform_v1beta1.services.tensorboard_service import (
+    TensorboardServiceClient,
+)
+from google.cloud.aiplatform_v1beta1.services.tensorboard_service.transports import (
+    grpc as transports_grpc,
+)
+
+_AUTH_SCOPE = "https://www.googleapis.com/auth/cloud-platform"
+_TB_GCP_RESOURCE_NAME = "projects/{}/locations/{}"
+_ENV_SPECIFIC_CONFIGS = {
+    "autopush": {
+        "api_server_hostname": "autopush-aiplatform.sandbox.googleapis.com",
+        "web_server_hostname": "tensorboard-gcp-dev.wm.r.appspot.com",
+    },
+    "staging": {
+        "api_server_hostname": "staging-aiplatform.sandbox.googleapis.com",
+        "web_server_hostname": "tensorboard-gcp-staging.uc.r.appspot.com",
+    },
+    "prod": {
+        "api_server_hostname": "aiplatform.googleapis.com",
+        "web_server_hostname": "tensorboard-gcp-prod.uc.r.appspot.com",
+    },
+}
+
+FLAGS = flags.FLAGS
+flags.DEFINE_string("experiment_name", None, "The name of the Cloud AI Experiment.")
+flags.DEFINE_string(
+    "experiment_display_name", None, "The display name of the Cloud AI Experiment."
+)
+flags.DEFINE_string("logdir", None, "Tensorboard log directory to upload")
+flags.DEFINE_bool("one_shot", False, "Iterate through logdir once to upload.")
+flags.DEFINE_string("api_uri", None, "DEPRECATED - API uri.")
+flags.DEFINE_string("env", "prod", "Environment which this tensorboard belongs to.")
+flags.DEFINE_string(
+    "tensorboard_resource_name",
+    None,
+    "Tensorboard resource to create this experiment in. ",
+)
+flags.DEFINE_integer(
+    "event_file_inactive_secs",
+    None,
+    "Age in seconds of last write after which an event file is considered " "inactive.",
+)
+flags.DEFINE_string(
+    "run_name_prefix",
+    None,
+    "If present, all runs created by this invocation will have their name "
+    "prefixed by this value.",
+)
+
+flags.DEFINE_multi_string(
+    "allowed_plugins",
+    [
+        scalar_metadata.PLUGIN_NAME,
+        histogram_metadata.PLUGIN_NAME,
+        distribution_metadata.PLUGIN_NAME,
+        text_metadata.PLUGIN_NAME,
+        hparams_metadata.PLUGIN_NAME,
+        images_metadata.PLUGIN_NAME,
+        graphs_metadata.PLUGIN_NAME,
+    ],
+    "Plugins allowed by the Uploader.",
+)
+
+flags.mark_flags_as_required(["experiment_name", "logdir", "tensorboard_resource_name"])
+
+
+def main(argv):
+    if len(argv) > 1:
+        raise app.UsageError("Too many command-line arguments.")
+
+    scope = _AUTH_SCOPE
```

Comment:
The auth and regionalization lines can reuse our shared logic.

You should be able to instantiate the the service client like so:

```Python
from google.cloud import aiplatform

api_client = aiplatform.initializer.global_config.create_client(
  client_class = TensorboardServiceClient,
  location_override = region
)
```

Reply:
Done